src/states/common/base.rs

```diff
@@ -139,54 +138,24 @@ pub trait Base: Display {
         let transition = match event {
             BootstrappedTo { node } => self.handle_bootstrapped_to(node),
             BootstrapFailure => self.handle_bootstrap_failure(outbox),
-            ConnectedTo {
-                peer: Peer::Node(peer_addr),
-            } => self.handle_connected_to(peer_addr, outbox),
-            ConnectedTo {
-                peer: Peer::Client(peer_addr),
-            } => {
-                let client_event = Client::Connected { peer_addr };
-                outbox.send_event(From::from(client_event));
-                Transition::Stay
-            }
+            ConnectedTo { peer } => match peer {
+                Peer::Client(_) => Transition::Stay,
+                Peer::Node(peer_addr) => self.handle_connected_to(peer_addr, outbox),
```

Comment:
This is what I was talking about here https://github.com/maidsafe/quic-p2p/pull/95#discussion_r378197863

Reply:
Yes, indeed, I think it make more sense as a follow up PR. I don't think I'll have time to do it if we hope to merge it before end of thursday. I'll raise an issue for it.

Reply:
Raised as https://github.com/maidsafe/routing/issues/2053